debug/slides_source/debug.tex

```diff
@@ -0,0 +1,382 @@
+\begin{lstlisting}
+import theano
+import theano.tensor as T
+from theano.tests.breakpoint import PdbBreakpoint
+input, target = T.fvectors('xy')
+
+mse = (input - target) ** 2
+
+# Conditional breakpoint to be activated if the total MSE is higher
+# than 100. The breakpoint will monitor the inputs, targets as well
+# as the individual error values
+breakpointOp = PdbBreakpoint(``MSE too high'')
+condition = T.gt(mse.sum(), 100)
+mse, monitored_input, monitored_target = breakpointOp(
+    condition, mse, input, target)
+\end{lstlisting}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{Breakpoint Op: example}
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        basicstyle=\footnotesize,
+        xleftmargin=-1em
+        }
+\begin{lstlisting}
+# Compile the theano function
+fct = theano.function([input, target], mse)
+
+# Use the function
+print fct([10, 0], [10, 5]) # Will NOT activate the breakpoint
+print fct([0, 0], [10, 5]) # Will activate the breakpoint
+\end{lstlisting}
+
+\end{frame}
+
+\subsection{PrintOp}
+\begin{frame}[fragile]
+  \frametitle{PrintOp}
+Allow to print during execution
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        basicstyle=\footnotesize,
+        xleftmargin=-1em
+        }
+\begin{lstlisting}
+import theano
+x = theano.tensor.vector()
+o = theano.printing.Print(``a message'')(x)
+f = theano.function([x], o)
+d = f([3, 4])
+a message  __str__ = [ 3.  4.]
+\end{lstlisting}
+\end{frame}
+
+
+\subsection{Nan}
+\begin{frame}
+  \frametitle{Most Frequent NaN Causes}
+\begin{itemize}
+\item Hyper parameter (ex: learning rate)
+\item Initialization of weight vectors
+\item Numerical Stability
+\item Algorithm Related
+\end{itemize}
+Run in NanGuardMode, DebugMode, or MonitorMode
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{NanGuardMode}
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        basicstyle=\footnotesize,
+        xleftmargin=-1em
+        }
+Can check:
+\begin{itemize}
+\item Nan
+\item Inf
+\item Big value (greater than 1e10)
+\end{itemize}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{NanGuardMode: example}
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        basicstyle=\footnotesize,
+        xleftmargin=-1em
+        }
+\begin{lstlisting}
+x = T.dmatrix()
+w = theano.shared(numpy.random.randn(5, 7))
+y = T.dot(x, w)
+mode=NanGuardMode(nan_is_error=True,
+                  inf_is_error=True,
+                  big_is_error=True)
+fun = theano.function(
+    [x], y, mode=mode)
+infa = numpy.tile(
+    (numpy.asarray(100.) ** 1000000), (3, 5))
+fun(infa)
+\end{lstlisting}
+\end{frame}
+
+\subsection{Test Value}
+\begin{frame}[fragile]
+  \frametitle{Test Value}
+Set a value to symbolic variable and use it when building the graph.
```

Comment:
'Give a sample value'
'variables'
Use it to do what? This needs something about why you would want to give test values to variables and what it buys you.